### ISSUE TITLE
CMake Error while trying to build for Debian (Ubuntu 12.10)

### ISSUE THREAD

**Issue by tomafc330** [dyle71/wirecloud-webkit-demo#1]
Hi there, I'm getting this message on the cmake:

```
tchan@tchan-desktop:~/repo/p4f/examples/wirecloud-webkit-demo/build$ cmake -Wno-dev ../bin
```

Error is:

```
CMake Error at CMakeLists.txt:61 (QT4_GENERATE_MOC):
  Unknown CMake command "QT4_GENERATE_MOC".
```

I've followed instructions to get the dependencies:

```
tchan@tchan-desktop:~/repo/p4f/examples/wirecloud-webkit-demo$ sudo apt-get install build-essential g++ gcc cmake libqt4-dev libboost-all-dev doxygen
```

What else do I need? Thanks

Tom

**Comment by dyle71** [comment]
#!/bin/hi Tom

On 08/10/2013 07:36 PM, tommytcchan wrote:

> Hi there, I'm getting this message on the cmake:
> 
> |tchan@tchan-desktop:~/repo/p4f/examples/wirecloud-webkit-demo/build$ cmake -Wno-dev ../bin
> |
> 
> Error is:
> 
> |CMake Error at CMakeLists.txt:61 (QT4_GENERATE_MOC):
>    Unknown CMake command "QT4_GENERATE_MOC".
> |
> 
> I've followed instructions to get the dependencies:
> 
> |tchan@tchan-desktop:~/repo/p4f/examples/wirecloud-webkit-demo$ sudo apt-get install build-essential g++ gcc cmake libqt4-dev libboost-all-dev doxygen
> |
> 
> What else do I need? Thanks

The QT4_GENERATE_MOC macro is part of the CMake installment for Qt4. 
Depending on your distribution this might come along cmake or with the 
developer versions of Qt4. I donnow.

The line given in the README is suitable for Debian 7.0 but your Ubuntu 
12.04 might handle things differently.

On my Gentoo machine I have:

$ grep -A 4 QT4_GENERATE_MOC /usr/share/cmake/Modules/FindQt4.cmake

# macro QT4_GENERATE_MOC(inputfile outputfile )

# creates a rule to run moc on infile and create outfile.

# Use this if for some reason QT4_WRAP_CPP() isn't appropriate, e.g.

# because you need a custom filename for the moc file or

something similar.
#
$ qfile /usr/share/cmake/Modules/FindQt4.cmake
dev-util/cmake (/usr/share/cmake/Modules/FindQt4.cmake)
$ eix -e cmake
[I] dev-util/cmake
      Available versions:  2.8.10.2-r2 (~)2.8.11.1 {emacs ncurses qt4 
qt5 test vim-syntax}
      Installed versions:  2.8.11.1(07:45:44 PM 07/27/2013)(ncurses qt4 
-emacs -qt5 -test -vim-syntax)
      Homepage:            http://www.cmake.org/
      Description:         Cross platform Make

However, I'm quite curious, because since when you run "cmake .." in the 
suggested build folder, it should include and check this FindQt4.cmake 
file and this ought to declare this macro.

Can you clean up the build folder and rerun a full cmake on this and 
paste the output?

KR:-O-

**Comment by dyle71** [comment]
Closing (for lost in time / missing feedback).
